addons/hr_calendar/tests/common.py

```diff
@@ -0,0 +1,123 @@
+        cls.calendar_28h_company_B.company_id = cls.company_B
+
+        cls.calendar_35h_night = cls.env['resource.calendar'].create({
+            'company_id': cls.company_A.id,
+            'tz': "Europe/Brussels",
+            'name': 'night calendar',
+            'attendance_ids': [
+                (0, 0, {'name': 'Monday Evening', 'dayofweek': '0', 'hour_from': 15, 'hour_to': 22, 'day_period': 'afternoon'}),
+                (0, 0, {'name': 'Tuesday Evening', 'dayofweek': '1', 'hour_from': 15, 'hour_to': 22, 'day_period': 'afternoon'}),
+                (0, 0, {'name': 'Wednesday Evening', 'dayofweek': '2', 'hour_from': 15, 'hour_to': 22, 'day_period': 'afternoon'}),
+                (0, 0, {'name': 'Thursday Evening', 'dayofweek': '3', 'hour_from': 15, 'hour_to': 22, 'day_period': 'afternoon'}),
+                (0, 0, {'name': 'Friday Evening', 'dayofweek': '4', 'hour_from': 15, 'hour_to': 22, 'day_period': 'afternoon'})
+            ],
+        })
+
+        cls.calendar_35h_night_company_B = cls.calendar_35h_night.copy()
+        cls.calendar_35h_night_company_B.company_id = cls.company_B
+
+        cls.partnerA = cls.env['res.partner'].create({
+            'name': "Partner A"
+        })
+        cls.partnerB = cls.env['res.partner'].create({
+            'name': "Partner B"
+        })
+        cls.partnerC = cls.env['res.partner'].create({
+            'name': "Partner C"
+        })
+        cls.partnerD = cls.env['res.partner'].create({
+            'name': "Partner D"
+        })
+
+        cls.employeeA = cls.env['hr.employee'].create({
+            'name': "Partner A - Calendar 35h",
+            'tz': "Europe/Brussels",
+            'resource_calendar_id': cls.calendar_35h.id,
+            'work_contact_id': cls.partnerA.id,
+            })
+        cls.employeeA_company_B = cls.env['hr.employee'].create({
+            'name': "Partner A - Calendar 28h",
+            'tz': "Europe/Brussels",
+            'resource_calendar_id': cls.calendar_28h_company_B.id,
+            'work_contact_id': cls.partnerA.id,
+            'company_id': cls.company_B.id
+            })
+
+        cls.employeeB = cls.env['hr.employee'].create({
+            'name': "Partner B - Calendar 28h",
+            'tz': "Europe/Brussels",
+            'resource_calendar_id': cls.calendar_28h.id,
+            'work_contact_id': cls.partnerB.id,
+            })
+        cls.employeeC = cls.env['hr.employee'].create({
+            'name': "Partner C - Calendar 35h night",
+            'tz': "Europe/Brussels",
+            'resource_calendar_id': cls.calendar_35h_night.id,
+            'work_contact_id': cls.partnerC.id,
+        })
+        cls.employeeD = cls.env['hr.employee'].create({
+            'name': "Partner D - Calendar 35h No contract",
+            'tz': "Europe/Brussels",
+            'resource_calendar_id': cls.calendar_35h.id,
+            'work_contact_id': cls.partnerD.id,
+            })
```

Comment:
nitpicking, same above
```suggestion
        })
```